index.js

```diff
@@ -143,27 +135,21 @@ function WriteStream(context, options) {
   this.endOffset = (options.end == null) ? Infinity : +options.end;
   this.bytesWritten = 0;
   this.pos = this.start;
-  this.destroyed = false;
-
-  this.on('finish', this.destroy.bind(this));
 }
 
 WriteStream.prototype._write = function(buffer, encoding, callback) {
   var self = this;
-  if (self.destroyed) return;
```

Comment:
@ronag The same question as earlier, you removed this check but on `ReadStream` you kept it, is that right?